### ISSUE TITLE
BFGS gets lost in Poisson regression example

### ISSUE THREAD

**Issue by user799595** [google/jax#17159]
### Description

I'm working on a Poisson regression example. I define the objective in JAX and then optimize it in two different ways:
1. With jax.scipy.optimize
2. With scipy.optimize

I tried to keep the two setups as similar as possible.

The JAX implementation seems to get stuck with a line search error (maybe related https://github.com/google/jax/issues/4594) and quite a large gradient.

Scipy terminates successfully.

```python
import numpy as np
import scipy.optimize
import jax
import jax.numpy as jnp
import jax.scipy.optimize
jax.config.update("jax_enable_x64", True)

# data generation
rng = np.random.default_rng(27)
T = 200
ts = np.linspace(0, 1, T+1)[:-1]
alpha = 0.1
beta = 1
rates = np.exp(alpha + beta * ts)
counts = rng.poisson(lam=rates)

# problem set up
X = jnp.asarray(np.c_[np.ones_like(ts), ts])
y = jnp.asarray(counts)
initial_params = jnp.zeros(X.shape[1])

def objective(params):
    log_rate = X @ params
    return jnp.sum(jnp.exp(log_rate) - y * log_rate)

# optimize using jax
print('jax.scipy.optimize results')
print(jax.scipy.optimize.minimize(objective, initial_params, method='bfgs', options={'line_search_maxiter': 1000, 'maxiter': 1000}))

# optimize using scipy
def wrapped_objective(params):
    params_jax = jnp.asarray(params)
    return np.float64(objective(params_jax))

def wrapped_gradient(params):
    params_jax = jnp.asarray(params)
    return np.asarray(jax.grad(objective)(params_jax))

print('scipy.optimize results')
print(scipy.optimize.minimize(wrapped_objective, np.asarray(initial_params), jac=wrapped_gradient, method='bfgs'))
```

Output
```
jax.scipy.optimize results
OptimizeResults(x=Array([0.50331112, 0.32459955], dtype=float64), success=Array(False, dtype=bool), status=Array(3, dtype=int64, weak_type=True), fun=Array(108.62462671, dtype=float64), jac=Array([-18.4706947 , -29.45663606], dtype=float64), hess_inv=Array([[ 0.23711728, -0.42412468],
       [-0.42412468,  0.77024769]], dtype=float64), nfev=Array(3, dtype=int64, weak_type=True), njev=Array(3, dtype=int64, weak_type=True), nit=Array(1, dtype=int64, weak_type=True))
scipy.optimize results
  message: Optimization terminated successfully.
  success: True
   status: 0
      fun: 102.37793361220376
        x: [ 2.246e-01  9.166e-01]
      nit: 8
      jac: [-2.120e-06 -1.385e-06]
 hess_inv: [[ 1.249e-02 -1.750e-02]
            [-1.750e-02  3.063e-02]]
     nfev: 12
     njev: 12
```

### What jax/jaxlib version are you using?

0.4.14

### Which accelerator(s) are you using?

CPU

### Additional system info

3.10.12, MacOS 13.5

**Comment by jakevdp** [comment]
Thanks for the report. We're not doing much maintenance on `jax.scipy.optimize`, and long-term we're planning to remove it. For optimization in JAX, I'd suggest checking out the [jaxopt](https://jaxopt.github.io/)  package.

**Comment by user799595** [comment]
Ah thanks, I hadn't heard of that package before.

**Comment by tavin** [comment]
jaxopt is a dead project now too, and optax doesn't have this capability...

Anyway if jax.scipy.optimize.minimize is known to be broken, the fix in https://github.com/google/jax/issues/16236 will not be merged, and it will someday be removed -- maybe the [documentation](https://jax.readthedocs.io/en/latest/_autosummary/jax.scipy.optimize.minimize.html) could be updated to say so?
